sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs

```diff
@@ -18,28 +18,73 @@ public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel i
             if (kind == CommunicationIdentifierKind.CallingApplication)
                 return new CallingApplicationIdentifier(AssertNotNull(id, nameof(identifier.Id), kind));
             if (kind == CommunicationIdentifierKind.PhoneNumber)
-                return new PhoneNumberIdentifier(AssertNotNull(identifier.PhoneNumber, nameof(identifier.PhoneNumber), kind));
+                return new PhoneNumberIdentifier(AssertNotNull(identifier.PhoneNumber, nameof(identifier.PhoneNumber), kind), AssertNotNull(id, nameof(identifier.Id), kind));
             if (kind == CommunicationIdentifierKind.MicrosoftTeamsUser)
             {
                 return new MicrosoftTeamsUserIdentifier(
-                        AssertNotNull(identifier.MicrosoftTeamsUserId, nameof(identifier.MicrosoftTeamsUserId), kind),
-                        AssertNotNull(identifier.IsAnonymous, nameof(identifier.IsAnonymous), kind));
+                    AssertNotNull(identifier.MicrosoftTeamsUserId, nameof(identifier.MicrosoftTeamsUserId), kind),
+                    AssertNotNull(identifier.IsAnonymous, nameof(identifier.IsAnonymous), kind),
+                    AssertNotNull(id, nameof(identifier.Id), kind),
+                    Deserialize(AssertNotNull(identifier.Cloud, nameof(identifier.Cloud), kind)));
             }
 
             return new UnknownIdentifier(AssertNotNull(id, nameof(identifier.Id), kind));
         }
 
+        private static CommunicationCloudEnvironment Deserialize(CommunicationCloudEnvironmentModel cloud)
+        {
+            if (cloud == CommunicationCloudEnvironmentModel.Public)
+                return CommunicationCloudEnvironment.Public;
+            if (cloud == CommunicationCloudEnvironmentModel.Gcch)
+                return CommunicationCloudEnvironment.Gcch;
+            if (cloud == CommunicationCloudEnvironmentModel.Dod)
+                return CommunicationCloudEnvironment.Dod;
+
+            throw new NotSupportedException();
```

Comment:
That enum doesn't seem very extensible to me, when we throw here...

Shouldn't we return `new CommunicationCloudEnvironment(CommunicationCloudEnvironmentModel.ToString())`?